### PR TITLE
Feat: sso-react package

### PR DESCRIPTION
**Pull request by pbastia**
A package providing a couple of front-end components and documentation, meant to be used with [sso-express](https://www.npmjs.com/package/@bcgov-cas/sso-express)

- [x] Session timeout handler
- [x] Logout warning modal
- [x] Session Refresher component
- [x] examples for login, logout and register buttons
- [x] documentation

### CODE REVIEW

File: packages/sso-react/src/SessionRefresher.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import throttle from "lodash.throttle";
+
+interface Props {
+  refreshUrl: string;
+  throttledTime?: number;
+  refreshEvents?: string[];
+}
+
+const SessionRefresher: React.FunctionComponent<Props> = ({
+  refreshUrl,
```
Comment (matthieu-foucault) [inline packages/sso-react/src/SessionRefresher.tsx:11]: should default to `/extend-session`

File: packages/sso-react/README.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+| refreshEvents | `string[]` | `["keydown", "mousedown", "scroll"]` | Events that trigger a session refresh |
+
+- Example: Refresh session on keydown, at most once every 10 minutes
+
+```tsx
+function App() {
+  return (
+    <div>
+      <SessionRefresher
+        refreshUrl="/session-idle-remaining-time"
```
Comment (matthieu-foucault) [inline packages/sso-react/README.md:27]:
```suggestion
        refreshUrl="/extend-session"
```

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]

File: packages/sso-react/README.md (hunk 2 above)
Comment (pbastia) [reply packages/sso-react/README.md:27]: good catch

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]

File: packages/sso-react/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    "declaration": true,
+    "moduleResolution": "node",
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "esModuleInterop": true,
+    "noImplicitReturns": true,
+    "noImplicitThis": true,
+    "suppressImplicitAnyIndexErrors": true,
+    "allowSyntheticDefaultImports": true,
+    "types": ["jest", "jest-extended"]
```
[the remark this answers is not in the document]
Comment (pbastia) [reply packages/sso-react/tsconfig.json:17]: we do for the `yarn test` directive to work, that other tsconfig file is for VSCode only

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]
Just a couple of documentation updates needed

**Review by matthieu-foucault — APPROVED** [review]
Merging this prior to `sso-express` rewrite

**Pull request merged**